smartsim/_core/utils/network.py

```diff
@@ -86,3 +87,13 @@ def current_ip(interface: str = "lo") -> str:  # pragma: no cover
         return get_ip_from_interface(loopback)
 
     return get_ip_from_interface(interface)
+
+
+def get_best_interface_and_address() -> t.Tuple[t.Optional[str], t.Optional[str]]:
+    available_ifs = psutil.net_if_addrs()
+    # TODO make this a CONFIG-time parameter
+    known_ifs = ["hsn", "ipogif", "ib"]
+    for interface in available_ifs:
+        if any(interface.startswith(if_prefix) for if_prefix in known_ifs):
+            return interface, get_ip_from_interface(interface)
+    return None, None
```

Comment:
Could we return an actual object instead of a tuple, or even a named tuple. There is no indication of which position in the resulting tuple is the address and which is the interface in this style.

Reply:
Yeah, I felt like it was half baked. Named tuple it is.